scrunch/tests/test_datasets.py

```diff
@@ -1777,15 +1777,64 @@ def _create(*args):
             'is_published': False,
         }))
 
+
+    def test_fork_project(self):
+        project_id = 'https://test.crunch.io/api/projects/1234/'
+        sess = MagicMock()
+        body = JSONObject({
+            'name': 'ds name',
+            'description': 'ds description',
+            'owner': project_id,
+        })
+        ds_res = MagicMock(session=sess, body=body)
+        ds_res.forks = MagicMock()
+        ds_res.forks.index = {}
+        ds = BaseDataset(ds_res)
+
+        expected_payload = {
+            'element': 'shoji:entity',
+            'body': {
+                'name': 'FORK #1 of ds name',
+                'description': 'ds description',
+                'owner': project_id,  # Project added
+                'is_published': False,
+            }
+        }
+        # Using project parameter
+        ds.fork(preserve_owner=False, project=project_id)
+        ds_res.forks.create.assert_called_with(expected_payload)
+
+        # Using owner parameter
+        ds.fork(preserve_owner=False, owner=project_id)
+        ds_res.forks.create.assert_called_with(expected_payload)
+
+        # Validations
+        err_msg1 = (
+            "Cannot pass both 'project' & 'owner' parameters together. "
+            "Please try again by passing only 'project' parameter."
+        )
+        with pytest.raises(ValueError, match=err_msg1):
+            ds.fork(owner="ABCD", project="1234")
+        
+        err_msg2 = ("Cannot pass 'project' or 'owner' when preserve_owner=True")
+        with pytest.raises(ValueError, match=err_msg2):
+            ds.fork(preserve_owner=True, project="1234")
+            ds.fork(preserve_owner=True, owner="1234")
```

Comment:
@shaikh-ma I am not sure about this test. Could you try to rewrite it like this?

```suggestion
        err_msg2 = ("Cannot pass 'project' or 'owner' when preserve_owner=True")
        with pytest.raises(ValueError, match=err_msg2):
            ds.fork(preserve_owner=True, project="1234")
            
         with pytest.raises(ValueError, match=err_msg2):
            ds.fork(preserve_owner=True, owner="1234")
```

so that you can test the different scenatios? I am just not sure that the check you wrote 
```
# Setting project value based on presere_owner
        if preserve_owner and project:
            raise ValueError("Cannot pass 'project' or 'owner' when preserve_owner=True")
```
does what expected. But if the test passes, then it's good. Thanks

Reply:
Done